python/cudf/cudf/core/scalar.py

```diff
@@ -14,9 +14,10 @@
     get_allowed_combinations_for_operator,
     to_cudf_compatible_scalar,
 )
+from cudf.utils.utils import CachedInstanceMeta
 
 
-class Scalar(BinaryOperand):
+class Scalar(BinaryOperand, metaclass=CachedInstanceMeta):
```

Comment:
This is a good question, I feel that no matter what value we choose one can always imagine some workflow that would see a weird performance change at the maxsize. What are the implications of using a very large number? I can see we're looking up via a dict in the end, so that part doesn't seem too bad at large numbers at least. There could I suppose be overhead from keeping a large number of python objects alive which take up memory. Each one could potentially also be holding onto a device pointer, although I'm a little fuzzier on what kind of issues that could cause (maybe some kind of serialized freeing like we've seen elsewhere?)

Reply:
I'm not a 100% sure of the implications either. Pool memory fragmentation sounds like one potential issue with a very large `maxsize`?

Note that there is a separate question of cached instances potentially keeping memory pools alive discussed [here](https://github.com/rapidsai/cudf/pull/11246#discussion_r919317686). But that is unrelated to `maxsize`, as  even just _one_ cached instance could keep an entire memory pool alive.